### ISSUE TITLE
Splat Expressions giving weird results?

### ISSUE THREAD

**Issue by PrzemyslawKlys** [SeeminglyScience/EditorServicesCommandSuite#50]
Hi @SeeminglyScience 

So I have this:

![image](https://user-images.githubusercontent.com/15063294/75109673-cbba8a00-5625-11ea-9dde-ab2356fea1e8.png)

I choose splat expression.. and:

1. If the cursor is inside Filter it will do this:

![image](https://user-images.githubusercontent.com/15063294/75109689-f86ea180-5625-11ea-874d-1032831276d4.png)

2. If the cursor is on Get-ADUser it will do this:

![image](https://user-images.githubusercontent.com/15063294/75109697-0e7c6200-5626-11ea-940c-3a729df931ab.png)

```powershell
$UsersActiveDirectory = $getADUserSplat = @{
    Filter = { Enabled -eq $true }
    SearchBase = $SearchBase
    Properties = 'ProxyAddresses', 'EmailAddress', 'DistinguishedName', 'Enabled', 'GivenName', 'Name', 'SamAccountName', 'SID', 'ObjectGUID'
}
Get-ADUser @getADUserSplat
```

Something is wrong or I am just doing it wrong :-) Also in some other situation it was doing this:

![image](https://user-images.githubusercontent.com/15063294/75098782-c5db8f00-55ba-11ea-9b45-63ed61042232.png)

I picked all parameters instead of just simple splat

```powershell
                    $newHTMLStatusItemSplat = @{
                        Icon = 'Bad'
                        Percentage = '30%'
                        ServiceName = $stringServiceName
                        ServiceStatus = $stringServiceStatus
                        Name = 'Twitter'
                        Status = 'Who knows?!'
                    }
                    New-HTMLStatusItem @newHTMLStatusItemSplat
```

But actually Name = ServiceName, ServiceStatus = Status now and it's just aliased.

![image](https://user-images.githubusercontent.com/15063294/75098801-e6a3e480-55ba-11ea-9aeb-42e6e1897eea.png)

But after I've rebuilt the module it was working ok, yet it was picking up both ServiceName and Name at the same time from somewhere.

I have a feeling it wasn't working like that before. Now it's a bit unpredictable. But maybe I wasn't using it enough ;)

**Comment by SeeminglyScience** [comment]
So the first part is working as expected.  From the perspective of static analysis, `Enabled` is a command and `-eq` is a parameter name. Really `Get-ADUser`'s `-Filter` parameter isn't typed as a `ScriptBlock`.  Instead it's converted to a string and the command internally does some hacky string scanning to look for variables.  Most folks (including myself) will recommend rather strongly against using the psuedo scriptblock style (even if documentation uses it).  There are *tons* of scenarios where it just straight up doesn't work like a scriptblock.

For the second example, did you have version of the module without that parameter alias in your module path?  I ran a little test:

```powershell
function Test-Command {
    [CmdletBinding()]
    param(
        [Parameter()]
        [Alias('what')]
        [string] $Hi
    )
}

Test-Command -what 'eyy'
```

If I highlight the function and <kbd>F8</kbd> then it correctly resolves `what` to `Hi`.  If I remove the `Alias` and run that, it shows how you describe.  If I don't have the command defined at all, it doesn't work (as expected).

This feature relies heavily on command discovery,  so whatever gets resolved first will win.  The engine doesn't have support for generating a `CommandInfo` purely from an AST, and manually building parameter set information from the AST isn't super feasible.

So really the solution to the latter is to import the module you're working on when you make changes like that.  If you manually load the command, it won't pull from module path.

**Comment by PrzemyslawKlys** [comment]
What about this conversion?

```
$DomainControllers = Get-WinADDomainControllers -TestAvailability
```

To

```powershell
$DomainControllers = $getWinADDomainControllersSplat = @{
    TestAvailability = $true
}
Get-WinADDomainControllers @getWinADDomainControllersSplat
```

It's behaving like that now. $DomainControllers should be elsewhere...

**Comment by SeeminglyScience** [comment]
Yeah that's wrong for sure 🙂

Working on it now, thanks!

**Comment by SeeminglyScience** [comment]
Fixed with v1.0.0-beta1

```powershell
Install-Module EditorServicesCommandSuite -Scope CurrentUser -AllowPrerelease -Force
```
